demo/lib/example_target.rb

```diff
@@ -13,25 +13,21 @@
 require 'cosmos/tools/cmd_tlm_server/interface_thread'
 
 module Cosmos
-
   class ExampleTarget
-
     class ExampleServerInterface < TcpipServerInterface
       PORT = 9999
 
       def initialize
         super(PORT, PORT, 5.0, nil, 'LENGTH', 0, 32, 4, 1, 'BIG_ENDIAN', 4)
       end
 
-      def pre_write_packet(packet)
-        [packet.length].pack('N') << packet.buffer(false)
+      def pre_write_data(packet_data)
+        packet_data.prepend([packet_data.length].pack('N'))
```

Comment:
This is a change to the API. pre_write_data now takes a buffer and returns a buffer.